R/sampler.R

```diff
@@ -249,3 +248,51 @@ scale_log_weights <- function(log_weights) {
   }
   list(weights = weights, average = average)
 }
+
+
+pre_data_run <- function(model, data, seeding_func,
+                         n_particles, i_state, particles){
+  
+  steps_per_day <- attr(data, "steps_per_day")
+  
+  state <- array(NA_real_,c(length(i_state),n_particles))
+  
+  seeding <- seeding_func(n_particles)
+  
+  if (!is.null(particles)) {
```

Comment:
In fact, I think you should get rid of the section in `particle_filter()` which sets up `particles` and move the array initialisation here. Get rid of the `particles` input to this function, add `save_particles` and return either `NULL` or the array depending on whether true or false

Reply:
Done